src/pudl/extract/eiaaeo.py

```diff
@@ -0,0 +1,453 @@
+    that may not be shared across multiple timeseries.
+    """
+
+    series_id: str
+    name: str
+    last_updated: str
+    units: str | None = None
+    data: list[tuple[str, str | float]]  # time, value
+
+
+class AEOTable(DataFrameModel):
+    """Data schema for a raw AEO table."""
+
+    date: Timestamp = Field(coerce=True)
+    value: str = Field(coerce=True)
+    units: str = Field(coerce=True)
+    series_name: str = Field(coerce=True)
+    category_name: str = Field(coerce=True)
+    case: str = Field(coerce=True)
+
+
+class AEOTaxonomy:
+    """Container for *all* the information in one AEO report.
+
+    AEO reports are composed of *categories*, which are metadata about multiple
+    data series, and *series*, which are the actual data + metadata associated
+    with one specific time series.
+
+    The categories and series form a DAG structure with 5 generations: root,
+    case, subject, leaf category, and data series.
+
+    The first generation is the root - there is one root node which is nameless
+    and which all other nodes descend from.
+
+    The second generation is the "cases." Cases are different scenarios within
+    the AEO. These have names like "Reference case," "High Economic Growth",
+    "Low Oil and Gas Supply." All direct children of the root node are cases.
+
+    The third generation is the "subjects." These are high-level tags, with
+    names like "Energy Prices", "Energy Consumption", etc. These are largely
+    used for filtering in the AEO data UI, so we ignore these.
+
+    The fourth generation is the "leaf categories." These are named things like
+    "Table 54.  Electric Power Projections by Electricity Market Module Region,
+    United States" and have a long list of "child series" which actually
+    contain the data. In other words, these leaf categories map the notion of
+    an AEO "table" to the actual data.
+
+    The fifth generation is the "data series." These actually contain the data
+    points, and have no children. They have names like "Electricity : Electric
+    Power Sector : Cumulative Planned Additions : Coal" and "Coal Supply :
+    Delivered Prices : Electric Power." As you can see the names imply a bunch
+    of different dimensions, which we don't try to make sense of in the extract
+    step.
+
+    In the first four generations we see a strictly branching tree, but many
+    leaf categories can point at the same data series so the whole taxonomy is
+    a DAG. This is because of two reasons:
+
+    * the subject tag doesn't affect data values, but because of the tree
+    structure, each leaf category is repeated once for each subject, leading to
+    multiple *duplicated* leaf categories pointing at the same data series.
+    * some data series are relevant to multiple different tables - so multiple
+    different leaf categories point at the same data series. In this case we
+    would expect the names of the leaf category to reflect their different
+    identities.
```

Comment:
```suggestion
    * the subject tag doesn't affect data values, but because of the tree
      structure, each leaf category is repeated once for each subject, leading to
      multiple *duplicated* leaf categories pointing at the same data series.
    * some data series are relevant to multiple different tables - so multiple
      different leaf categories point at the same data series. In this case we
      would expect the names of the leaf category to reflect their different
      identities.
```